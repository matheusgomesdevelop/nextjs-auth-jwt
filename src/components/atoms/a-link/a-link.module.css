.a_link {
    color: var(--primary-color);
    text-decoration: none;
}
