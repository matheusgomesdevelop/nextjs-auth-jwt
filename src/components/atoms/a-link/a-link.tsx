import { ReactNode } from 'react';

// StyleSheet
import styles from './a-link.module.css';

interface I_ALink {
    href: string;
    children: ReactNode;
}

export const A_Link = ({ href, children }: I_ALink) => {
    return (
        <a href={href} className={styles.a_link}>
            {children}
        </a>
    );
};
