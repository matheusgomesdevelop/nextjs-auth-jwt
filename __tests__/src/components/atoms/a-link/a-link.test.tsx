import renderer from 'react-test-renderer';

// Atoms
import { A_Link } from '@/components/atoms/a-link/a-link';

describe('Deve renderizar os links corretamente', () => {
    it('Deve preservar a estrutura visual dos links', () => {
        const three = renderer.create(<A_Link href="/" label="texto do link" />).toJSON();

        expect(three).toMatchSnapshot();
    });
});
