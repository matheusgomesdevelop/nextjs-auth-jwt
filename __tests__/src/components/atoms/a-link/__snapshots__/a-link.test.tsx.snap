// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar os links corretamente Deve preservar a estrutura visual dos links 1`] = `
<a
  className="a_link"
  href="/"
/>
`;
